server/pkg/api/accountHandlers.go

```diff
@@ -38,8 +43,19 @@ func createAccount(ctx apiContext, w http.ResponseWriter, r *http.Request) {
 
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error: %s\n", err.Error())
-		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
-		return
+		if err.Error() != "" {
+			profile := Profile{err.Error()}
+			js, err := json.Marshal(profile)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			w.Write(js)
+			return
+		} else {
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
```

Comment:
If you export the constant messages you can nest this into the `if err != nil` and see if it's one of the messages you send, else send a 500